cmd/fake/main.go

```diff
@@ -0,0 +1,77 @@
+package main
+
+import (
+	"flag"
+	"fmt"
+	"math/rand"
+	"net/http"
+	"strconv"
+	"strings"
+	"time"
+)
+
+var counts, errorRate, errors int
+var errorCodes []int
+
+func replyOK(w http.ResponseWriter, r *http.Request) {
+	message := strings.TrimPrefix("/", r.URL.Path)
+	message = message + " served ok"
+
+	responseCode := http.StatusOK
+	w.WriteHeader(responseCode)
+	w.Write([]byte(message))
+}
+
+func replyError(w http.ResponseWriter, r *http.Request) {
+	message := strings.TrimPrefix("/", r.URL.Path)
+	message = "Errored request to " + message
+
+	responseCode := errorCodes[rand.Intn(len(errorCodes))]
+	w.WriteHeader(responseCode)
+	w.Write([]byte(message))
+}
+
+func countRequests(w http.ResponseWriter, r *http.Request) {
+	counts = counts + 1
+	if errorRate > 0 && counts%(100/errorRate) == 0 {
+		errors = errors + 1
+		replyError(w, r)
+	} else {
+		replyOK(w, r)
+	}
+}
+
+func printCounts() {
+	for range time.Tick(10 * time.Second) {
+		fmt.Printf("requests=%v errors=%v\n", counts, errors)
+		counts = 0
+		errors = 0
+	}
+}
+
+func parseInts(ints *string) (intList []int) {
+	for _, n := range strings.Split(*ints, ",") {
+		value, err := strconv.Atoi(n)
+		if err != nil {
+			panic(err)
+		}
+		intList = append(intList, value)
+	}
+	return intList
+}
+
+func main() {
+	rand.Seed(time.Now().Unix())
```

Comment:
Do we really need to do this before parsing the flags?

Reply:
Now we don't need this anymore, since now the error codes are used sequential and repeated.